libraries/ms-helper/src/mcp2515.c

```diff
@@ -65,72 +77,102 @@ static uint8_t prv_read_status(Mcp2515Storage *storage) {
   return read_data[0];
 }
 
+// Message RX
 static void prv_handle_rx(Mcp2515Storage *storage, uint8_t int_flags) {
   for (size_t i = 0; i < SIZEOF_ARRAY(s_rx_buffers); i++) {
     Mcp2515RxBuffer *rx_buf = &s_rx_buffers[i];
     if (int_flags & rx_buf->int_flag) {
-      // Message RX
-
       // Read ID
       uint8_t id_payload[] = { MCP2515_CMD_READ_RX | rx_buf->id };
       Mcp2515IdRegs read_id_regs = { 0 };
       spi_exchange(storage->spi_port, id_payload, sizeof(id_payload), (uint8_t *)&read_id_regs,
                    sizeof(read_id_regs));
+
       Mcp2515Id id = {
         .sid_0_2 = read_id_regs.sidl.sid_0_2,
         .sidh = read_id_regs.sidh,
         .eid0 = read_id_regs.eid0,
         .eid8 = read_id_regs.eid8,
         .eid_16_17 = read_id_regs.sidl.eid_16_17,
       };
+
       bool extended = read_id_regs.sidl.ide;
       size_t dlc = read_id_regs.dlc.dlc;
 
       if (!extended) {
         // Standard IDs have garbage in the extended fields
-        id.raw &= 0x7FF;
+        id.raw >>= MCP2515_EXTENDED_ID_LEN;
       }
 
       uint8_t data_payload[] = { MCP2515_CMD_READ_RX | rx_buf->data };
       uint64_t read_data = 0;
       spi_exchange(storage->spi_port, data_payload, sizeof(data_payload), (uint8_t *)&read_data,
                    sizeof(read_data));
-
+      // Clear the interrupt flag so a new message can be loaded
+      prv_bit_modify(storage, MCP2515_CTRL_REG_CANINTF, rx_buf->int_flag, 0x0);
       if (storage->rx_cb != NULL) {
         storage->rx_cb(id.raw, extended, read_data, dlc, storage->context);
       }
     }
   }
 }
 
-static void prv_handle_error(Mcp2515Storage *storage, uint8_t int_flags) {
+static void prv_handle_error(Mcp2515Storage *storage, uint8_t int_flags, uint8_t err_flags) {
+  // Clear flags
   if (int_flags & MCP2515_CANINT_EFLAG) {
-    // Don't bother reporting errors - just clear them
-    uint8_t clear = 0x00;
+    // Clear error flag
+    LOG_DEBUG("MCP2515_CANINT_EFLAG error\n");
```

Comment:
these log_debugs will slow down the application, make sure the error callback gets called for any errors, instead of log_debugging...

Reply:
Removed and ensured the error callback gets called.